pkg/endpoint/endpoint_test.go

```diff
@@ -0,0 +1,36 @@
+package endpoint
+
+import (
+	"context"
+	"fmt"
+	"testing"
+)
+
+func TestListen(t *testing.T) {
+	var config Config
+	var ctx context.Context
+	config.Listener = ""
+	endpoints := [13][2]string{{"postgres://doadmin:ax4t3h9twe04fa7k@postgressdb-do-user-898766-0.a.db.ondigitalocean.com:2560/defaultdb", "invalid port number"},
```

Comment:
This is not the right approcah for unit testing for this function, basically we want to make sure that the function returns an error when an invalid address is passed, so there are basically 3 cases:

- DSN with no <driver>:// and this should return the new error you implemented
- DSN with <driver>:// and invalid address and this should return an error
- DSN with correct <driver>:// and address

Reply:
you dont have to have a working mysql and postgres to test the last case, we basically have these integration tests in k3s https://github.com/rancher/k3s/blob/master/scripts/test-setup-sonobuoy-mysql

cc @cjellick